scripts/gui.lua

```diff
@@ -590,7 +597,55 @@ function GUI.onTextChanged(event)
             GUI.updateTrackerConfig(playerSettings.gui.trackerInfo, selectedTracker)
             GUI.updateTrackerInfo(playerSettings.gui.trackerInfo, selectedTracker)
         end
-    end
+    elseif event.element.name == "tlbe-tracker-cityblock-size-x" then
+        local selectedTracker = playerSettings.trackers[playerSettings.guiPersist.selectedTracker]
+        selectedTracker.cityBlock.blockSize.x = math.floor(tonumber(event.element.text))
+        Tracker.recalculateCityBlock(selectedTracker)
+
+        GUI.updateTrackerConfig(playerSettings.gui.trackerInfo, selectedTracker)
+        GUI.updateTrackerInfo(playerSettings.gui.trackerInfo, selectedTracker)
+    elseif event.element.name == "tlbe-tracker-cityblock-size-y" then
+        local selectedTracker = playerSettings.trackers[playerSettings.guiPersist.selectedTracker]
+        selectedTracker.cityBlock.blockSize.y = math.floor(tonumber(event.element.text))
+        Tracker.recalculateCityBlock(selectedTracker)
+
+        GUI.updateTrackerConfig(playerSettings.gui.trackerInfo, selectedTracker)
+        GUI.updateTrackerInfo(playerSettings.gui.trackerInfo, selectedTracker)
+    elseif event.element.name == "tlbe-tracker-cityblock-offset-x" then
+        local selectedTracker = playerSettings.trackers[playerSettings.guiPersist.selectedTracker]
+        selectedTracker.cityBlock.blockOffset.x = math.floor(tonumber(event.element.text))
```

Comment:
I triggered an error on this line while testing:
```
Error while running event TLBE::on_gui_text_changed (ID 2)
__TLBE__/scripts/gui.lua:616: bad argument #1 of 2 to 'floor' (number expected, got nil)
stack traceback:
	[C]: in function 'floor'
	__TLBE__/scripts/gui.lua:616: in function <__TLBE__/scripts/gui.lua:511>
```

(field was empty as I was editing the value)

_Edit_ It is (ofc) for all fields (as they use the same method)

Reply:
I have fixed this for all events with a nil check.